synapse/rest/client/v1/room.py

```diff
@@ -230,15 +234,25 @@ async def on_POST(self, request, room_id, event_type, txn_id=None):
             "sender": requester.user.to_string(),
         }
 
-        if b"ts" in request.args and requester.app_service:
+        inherit_depth = False
+        prev_events = parse_strings_from_args(request.args, "prev_event")
+        if self._msc2716_enabled:
```

Comment:
This needs a `requester.app_service` check I believe.

Reply:
:+1: Had this discussed in https://github.com/matrix-org/synapse/pull/9247#discussion_r570745291 which probably fell off the most recent diff